CONTRIBUTING.md

```diff
@@ -1,123 +1,126 @@
+on setting up your environment.
 
-Once your change is ready, prepare the PR. The PR can contain any number of commits, but when it is merged, they will all get squashed. The commit names and descriptions can be arbitrary, but the name and the description of the PR must follow the following template:
+# Build Process
 
-```
-<type>: <name>
+Nearcore is a reasonably standard Rust project, so `cargo test` most likely will
+just work.  There are couple of specifics though:
+
+* `nearcore` assumes a UNIX-like environment.  Linux is actively supported; Mac
+is generally supported (the newer M1 chipset may not be well supported); and
+Windows is not supported.
+* `nearcore` build process includes generating bindings to RocksDB via
+`bindgen`, which requires `libclang`.  See [`bindgen`
+documentation](https://rust-lang.github.io/rust-bindgen/requirements.html#clang)
+for installation instructions.
+* At the moment, `nearcore` only supports compiling for x86_64 architecture (for
+technical reasons, we have to care about CPU architecture and supporting one is
+easier).  To compile to x86_64 on Apple silicon, use `rustup set default-host
+x86_64-apple-darwin` command.  That is, cross-compilation sadly doesn't work at
+the time of writing.
+* You can optionally use the system installations of `librocksdb`, `libsnappy`
+and `lz4` in order to speed up the compilation and reduce the build memory
+requirements by setting the `ROCKSDB_LIB_DIR`, `SNAPPY_LIB_DIR` and
+`LZ4_LIB_DIR` environment variables.  These environment variables should point at
+the directory containing the dynamic shared objects (`.so`s or `.dylib`s.)
+
+If your setup does not involve `rustup`, the required version of the rust
+toolchain can be found in the `rust-toolchain` file.
+
+# Pull Requests
+
+All the contributions to `nearcore` happen via Pull Requests.  To create a Pull
+Request, fork `nearcore`, create a new branch, do the work there, and then send
+a PR against the `master` branch.
+
+When working on a PR, please keep the following in mind:
+
+1. The changes should be thoroughly tested.  Please refer to [this
+document](https://github.com/nearprotocol/nearcore/wiki/Writing-tests-for-nearcore)
+for our testing guidelines and overview of the testing infrastructure.
+2. If you get an issue assigned to you, please post updates at least once a
+week.  You are encouraged to submit work-in-progress draft PRs to get early
+feedback.
+3. A PR can contain any number of commits and when merged, the commits will be
+squashed into a single commit.
+4. The PR name should follow the template: `<type>: <name>`.  
+Where `type` is `fix` for fixesö `feat` for new features; `refactor` for changes
+that primarily reorganize code; `doc` for changes that primarily change
+documentation or comments; `test` for changes that primarily introduce new
+tests; and `chore` for updating grunt tasks. Please note that `type` is case
+sensitive.
```

Comment:
Let's maybe itemize section titles for bettre readability: 

* `fix`
* `test`
* `feat`
* `chore`